src/handlers/gestures/useGesture.ts

```diff
@@ -0,0 +1,215 @@
+import React from 'react';
+import RNGestureHandlerModule from '../../RNGestureHandlerModule';
+import {
+  BaseGesture,
+  GestureRef,
+  GestureType,
+  HandlerCallbacks,
+  InteractionBuilder,
+} from './gesture';
+import { registerHandler, unregisterHandler } from '../handlersRegistry';
+import Animated from 'react-native-reanimated';
+import {
+  baseGestureHandlerWithMonitorProps,
+  filterConfig,
+} from '../gestureHandlerCommon';
+import { tapGestureHandlerProps } from '../TapGestureHandler';
+import {
+  panGestureHandlerProps,
+  panGestureHandlerCustomNativeProps,
+} from '../PanGestureHandler';
+import { longPressGestureHandlerProps } from '../LongPressGestureHandler';
+import { forceTouchGestureHandlerProps } from '../ForceTouchGestureHandler';
+import { flingGestureHandlerProps } from '../FlingGestureHandler';
+
+const ALLOWED_PROPS = [
+  ...baseGestureHandlerWithMonitorProps,
+  ...tapGestureHandlerProps,
+  ...panGestureHandlerProps,
+  ...panGestureHandlerCustomNativeProps,
+  ...longPressGestureHandlerProps,
+  ...forceTouchGestureHandlerProps,
+  ...flingGestureHandlerProps,
+];
+
+export type GestureConfigReference = {
+  config: GestureType[];
+  callback: null | (() => void);
+  animatedEventHandler: any;
+  animatedHandlers: Animated.SharedValue<
+    HandlerCallbacks<Record<string, unknown>>[] | null
+  > | null;
+  firstExecution: boolean;
+};
+
+export function useGesture(gestureConfig: InteractionBuilder | GestureType) {
+  const gesture = gestureConfig.configure();
+
+  const result = React.useRef<GestureConfigReference>({
+    config: gesture,
+    callback: null,
+    animatedEventHandler: null,
+    animatedHandlers: null,
+    firstExecution: true,
+  });
+
+  if (result.current.firstExecution) {
+    gestureConfig.initialize();
+  }
+
+  function dropHandlers() {
+    for (const g of result.current.config) {
+      RNGestureHandlerModule.dropGestureHandler(g.handlerTag);
+
+      unregisterHandler(g.handlerTag);
+    }
+  }
+
+  function convertToHandlerTag(ref: GestureRef): number {
+    if (typeof ref === 'number') {
+      return ref;
+    } else if (ref instanceof BaseGesture) {
+      return ref.handlerTag;
+    } else {
+      let tag = ref.current?.handlerTag;
+      if (tag === undefined) {
+        tag = -1;
+      }
+      return tag;
+    }
+  }
+
+  function attachHandlers() {
+    if (!result.current.firstExecution) {
+      gestureConfig.initialize();
+    } else {
+      result.current.firstExecution = false;
+    }
+
+    for (const gst of gesture) {
```

Comment:
Please rename.

Reply:
Done in b1535912f84e63f33bf3e83e92be2ebea6a5de84